index.js

```diff
@@ -164,172 +161,190 @@ function buildInfo(source) {
 }
 
 /*
- * Assigns host, basePath and schemes for Swagger 2.0 result document from
+ * Get host, basePath and schemes for Swagger 2.0 result document from
  * Swagger 1.2 basePath.
  * @param basePath {string} - the base path from Swagger 1.2
- * @param result {object} - Swagger 2.0 document
 */
-function assignPathComponents(basePath, result) {
+function buildPathComponents(basePath) {
+  if (!basePath) { return {}; }
+
   var url = urlParse(basePath);
-  result.host = url.host;
-  result.basePath = url.path;
-  if (url.protocol) {
-    result.schemes = [url.protocol.substr(0, url.protocol.length - 1)];
-  }
+  return {
+    host: url.host,
+    basePath: url.path || '/',
+    //url.protocol include traling colon
+    schemes: url.protocol && [url.protocol.slice(0, -1)]
+  };
+}
+
+/*
+ * Builds a Swagger 2.0 type properites from a Swagger 1.2 type properties
+ *
+ * @param oldDataType {object} - Swagger 1.2 type object
+ *
+ * @returns {object} - Swagger 2.0 equivalent
+ */
+function buildTypeProperties(oldType) {
+  if (!oldType) { return {}; }
+
+  //TODO: handle list[<TYPE>] types from 1.1 spec
+
+  var typeMap = {
+    integer:  {type: 'integer'},
+    number:   {type: 'number'},
+    string:   {type: 'string'},
+    boolean:  {type: 'boolean'},
+    array:    {type: 'array'},
+    file:     {type: 'file'},
+    int:      {type: 'integer', format: 'int32'},
+    long:     {type: 'integer', format: 'int64'},
+    float:    {type: 'number',  format: 'float'},
+    double:   {type: 'double',  format: 'double'},
+    byte:     {type: 'string',  format: 'byte'},
+    date:     {type: 'string',  format: 'date'},
+    datetime: {type: 'string',  format: 'date-time'},
+    list:     {type: 'array'},
+    void:     {}
+  };
+
+  return typeMap[oldType.toLowerCase()] || {$ref: oldType};
```

Comment:
Yes it's behaviour that I'm already observing on my samples.
I think correct solution will be to compile list of definitions names.
And search in that list before trying to convert primitive types.
And in case if search in `definitions` and `primitive types` both fails emit an error.

Problem is that I need to pass list of definitions names some how.
I want to convert library functions into methods of class in this case I can make list of name private member.
This will not affect any public API, only internal functions.
But I think this should be discussed separately.

Meanwhile, behaviour will stay the same as it was before my changes, see [this](https://github.com/apigee-127/swagger-converter/blob/master/index.js#L204)